libcontainer/container_linux.go

```diff
@@ -1822,14 +1822,23 @@ func (c *linuxContainer) runType() Status {
 }
 
 func (c *linuxContainer) isPaused() (bool, error) {
-	fcg := c.cgroupManager.GetPaths()["freezer"]
-	if fcg == "" {
-		// A container doesn't have a freezer cgroup
-		return false, nil
-	}
-	pausedState := "FROZEN"
-	filename := "freezer.state"
-	if cgroups.IsCgroup2UnifiedMode() {
+	var fcg, filename, pausedState string
+
+	if !cgroups.IsCgroup2UnifiedMode() {
+		fcg = c.cgroupManager.GetPaths()["freezer"]
+		if fcg == "" {
+			// A container doesn't have a freezer cgroup
+			return false, nil
+		}
+		filename = "freezer.state"
```

Comment:
Just curious: the filenames are different for the unified versus non-unified cases. I assume this is intentional.

Reply:
@tedyu yes, cgroups v2 have different filenames for various controllers.